### PR TITLE
[SG2] Main Preview Zoom + Pan/Rotate and support for undocking/resizing preview overlay

### PR DESCRIPTION
**Pull request by Nightmask3**
---
### Purpose of this PR

Adds ability to zoom in/out on main preview, rotate/pan the preview mesh around, and ensures that undocking/resizing the preview doesn't break it.

The preview rotation, zoom and mesh are now stored in the UserPrefs and will be persisted per graph asset, between graph sessions. 

https://user-images.githubusercontent.com/13033542/179117553-1bdeba6a-cb76-49e8-9b93-0d8647f63c62.mp4

Also did some cleanup and removed some old files that were lying around, so don't be dismayed by the large file count, if anything: 

image

---
### Testing status

Testrails:

1) Zooming in/out of preview: https://qatestrail.hq.unity3d.com/index.php?/cases/view/1081996
2) Rotating preview: https://qatestrail.hq.unity3d.com/index.php?/cases/view/1081996
3) Also updated the older mesh change testrail item to account for the mesh being serialized between graph sessions.

### CODE REVIEW

File: com.unity.graphtools.foundation/Editor/UI/Views/GraphView/GraphView.cs [hunk 1]
```diff
@@ -1631,12 +1631,11 @@ protected virtual void DeleteElementsFromChangeSet(GraphModelStateComponent.Chan
 
                 k_UpdateAllUIs.Clear();
 
-                // GTF-Edit: Commented this out because on deletion it tries to represent our port models and fails
                 // ToList is needed to bake the dependencies.
-                //foreach (var ui in model.GetDependencies().ToList())
-                //{
-                //    ui.UpdateFromModel();
-                //}
+                foreach (var ui in model.GetDependencies().ToList())
```
Comment (Nightmask3) [inline com.unity.graphtools.foundation/Editor/UI/Views/GraphView/GraphView.cs:1635]: this is a GTF-Edit that's not needed, and also wasn't added to the PR we just submitted

File: com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs [hunk 2]
```diff
@@ -464,6 +460,25 @@ public void RequestNodePreviewShaderCodeStrings(string nodeName, out ShaderMessa
...
+            // This will queue it up for re-render
+            m_MainPreviewData.isRenderOutOfDate = true;
+
+            // Main Preview window was resized, RTs cannot be resized so create a new one
+            m_MainPreviewData.renderTexture = new RenderTexture(
+                width,
+                height,
+                16,
+                RenderTextureFormat.ARGB32,
+                RenderTextureReadWrite.Default) { hideFlags = HideFlags.HideAndDontSave };
```
Comment (esmelusina) [inline com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs:477]: Nit: Formatting.

File: com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandsRegistrar.cs [hunk 3]
```diff
@@ -112,6 +56,40 @@ public static class ShaderGraphCommandsRegistrar
...
+            //    AddPortCommand.DefaultCommandHandler,
+            //    graphTool.UndoStateComponent,
+            //    graphView.GraphViewState
+            //);
+
+            //commandDispatcher.RegisterCommandHandler<UndoStateComponent, GraphModelStateComponent, RemovePortCommand>(
+            //    RemovePortCommand.DefaultCommandHandler,
+            //    graphTool.UndoStateComponent,
+            //    graphView.GraphModelState
+            //);
```
Comment (esmelusina) [inline com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandsRegistrar.cs:71]: Either delete commented code or put a longer explanation in for why we should keep it.

Comment (Nightmask3) [reply com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandsRegistrar.cs:71]: oh yeah I deleted those files and forgot to remove these, thanks for calling it out

File: com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs (hunk 2 above)
Comment (Nightmask3) [reply com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs:477]: changed it

**Review by esmelusina — APPROVED** [review]

**Pull request merged**